File: examples/governance_static_go/model_registry_static_go/domain/classified_risk.go
package domain

import "time"

type ClassifiedRisk struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	ModelId string `json:"model_id"`
	RiskLevel string `json:"risk_level"`
	Name string `json:"name"`
	Version string `json:"version"`
	ProviderId string `json:"provider_id"`
	Description string `json:"description"`
	RegisteredAt time.Time `json:"registered_at"`
	ParentModelId string `json:"parent_model_id"`
	DerivationType string `json:"derivation_type"`
	Capabilities []Capability `json:"capabilities"`
	IntendedUses []IntendedUse `json:"intended_uses"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e ClassifiedRisk) EventName() string { return "ClassifiedRisk" }

func (e ClassifiedRisk) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/model_registry_domain/lib/model_registry_domain/ai_model/events/classified_risk.rb
module ModelRegistryDomain
  class AiModel
    module Events
      class ClassifiedRisk
        attr_reader :aggregate_id, :model_id, :risk_level, :name, :version, :provider_id, :description, :registered_at, :parent_model_id, :derivation_type, :capabilities, :intended_uses, :status, :occurred_at

        def initialize(aggregate_id: nil, model_id: nil, risk_level: nil, name: nil, version: nil, provider_id: nil, description: nil, registered_at: nil, parent_model_id: nil, derivation_type: nil, capabilities: nil, intended_uses: nil, status: nil)
          @aggregate_id = aggregate_id
          @model_id = model_id
          @risk_level = risk_level
          @name = name
          @version = version
          @provider_id = provider_id
          @description = description
          @registered_at = registered_at
          @parent_model_id = parent_model_id
          @derivation_type = derivation_type
          @capabilities = capabilities
          @intended_uses = intended_uses
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
